src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -608,6 +608,12 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
 
         internal BoundLocalDeconstructionDeclaration BindDeconstructionDeclarationStatement(DeconstructionDeclarationStatementSyntax node, DiagnosticBag diagnostics)
         {
+            var constKeyword = node.Modifiers.FirstOrDefault(SyntaxKind.ConstKeyword);
+            if (constKeyword != default(SyntaxToken))
```

Comment:
@cston: Not knowing how the internals work, I would expect the error over the `const` simply because the `const` keyword will be the piece that needs changing.